xml/upgrade-1.3.xsl

```diff
@@ -1,61 +1,68 @@
-<?xml version="1.0" encoding="ISO-8859-1"?>
-<xsl:stylesheet version="1.0"
-		xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
-		xmlns:fn="http://www.w3.org/2005/02/xpath-functions">
-<xsl:output method='xml' version='1.0' encoding='UTF-8' indent='yes'/>
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+<xsl:output method="xml" encoding="UTF-8" indent="yes"/>
 
 <xsl:template match="role_ref">
   <xsl:element name="role">
-    <xsl:apply-templates select="@*"/>
-    <xsl:apply-templates select="node()" />
+    <xsl:apply-templates select="@*|node()"/>
   </xsl:element>
 </xsl:template>
 
 <xsl:template match="read|write|deny">
   <xsl:element name="acl_permission">
 
-    <xsl:attribute name="id"><xsl:value-of select="@id"/></xsl:attribute>
+    <xsl:copy-of select="@id"/>
     <xsl:attribute name="kind"><xsl:value-of select="name()"/></xsl:attribute>
 
-    <xsl:if test="@ref">
-      <xsl:attribute name="reference"><xsl:value-of select="@ref"/></xsl:attribute>
-    </xsl:if>
-    <xsl:if test="not(@ref)">
-      <xsl:if test="@tag">
+    <xsl:choose>
+      <xsl:when test="@ref">
+        <xsl:attribute name="reference"><xsl:value-of select="@ref"/></xsl:attribute>
+        <xsl:if test="@attribute">
+          <!-- alternatively, rephrase (generalized a bit) turning it to @xpath -->
+          <xsl:message>@attribute cannot accompany @ref for upgrade-1.3.xsl purposes, hence dropped</xsl:message>
```

Comment:
But she is expected to understand XPath if that's used in CIB configuration :)
Rephrasing just slightly.  At least some information about the flawed upgrade
path is better than nothing.